### PR TITLE
CGAL: Test the license check mechanism

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

In the Triangulation_3 testsuite I added a file with the `#define`s that switch on license checking and the `#define`s of the package and its dependencies.  
**Question:** Should the dependencies go into the folder package_info  ?

## Release Management

* Affected package(s): Currently Triangulation_3, but it will be generalized

### CODE REVIEW

**Comment by lrineau** [comment]
I am not sure `package_info/` is part of the internal releases tarballs, and even if it was, I think our testsuite scripts cannot know where to find it. Only the `test/` sub-directories are copied. So my opinion is that `dependencies.h` should not be in `package_info/`.

As it is, it might be really surprising for any CGAL developer not aware of the feature. I suggest to rename the file "test_dependencies.h", and include as a comment a link to the documentation, or a text that explains what is tested.

File: Triangulation_3/test/Triangulation_3/test_dependies.h [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# For the license check mechanism see:
+# http://doc.cgal.org/latest/Manual/preliminaries.html#title5
+
+# This package depends on the following packages
```
Comment (lrineau) [inline Triangulation_3/test/Triangulation_3/test_dependies.h]: You renamed the file with a typo in the new name.

Beside that error, the comment seems good.

**Comment by lrineau** [comment]
The test failed:
```
[ 50%] Building CXX object CMakeFiles/test_delaunay_3.dir/test_delaunay_3.cpp.o
/usr/bin/c++   -DCGAL_TEST_SUITE -DCGAL_USE_GMP -DCGAL_USE_MPFR -I/home/cgal_tester/build/src/cmake/platforms/Fedora-rawhide-Release/test/Triangulation_3/include -I/home/cgal_tester/build/src/cmake/platforms/Fedora-rawhide-Release/test/Triangulation_3/../../include -I/home/cgal_tester/build/src/cmake/platforms/Fedora-rawhide-Release/test/Triangulation_3 -I/home/cgal_tester/build/src/cmake/platforms/Fedora-rawhide-Release/include -I/mnt/testsuite/include  -Wall -frounding-math -O3 -DCGAL_NDEBUG   -o CMakeFiles/test_delaunay_3.dir/test_delaunay_3.cpp.o -c /home/cgal_tester/build/src/cmake/platforms/Fedora-rawhide-Release/test/Triangulation_3/test_delaunay_3.cpp
In file included from /mnt/testsuite/include/CGAL/Delaunay_triangulation_3.h:27:0,
                 from /home/cgal_tester/build/src/cmake/platforms/Fedora-rawhide-Release/test/Triangulation_3/test_delaunay_3.cpp:25:
/mnt/testsuite/include/CGAL/license/Triangulation_3.h:39:8: error: #error "Your commercial license for CGAL does not cover this release of the 3D Triangulations package. You get this error, as you defined CGAL_LICENSE_ERROR."
 #      error "Your commercial license for CGAL does not cover this release \
        ^~~~~
In file included from /mnt/testsuite/include/CGAL/triangulation_assertions.h:43:0,
                 from /mnt/testsuite/include/CGAL/Triangulation_3.h:43,
                 from /mnt/testsuite/include/CGAL/Delaunay_triangulation_3.h:40,
                 from /home/cgal_tester/build/src/cmake/platforms/Fedora-rawhide-Release/test/Triangulation_3/test_delaunay_3.cpp:25:
/mnt/testsuite/include/CGAL/license/Triangulation_2.h:39:8: error: #error "Your commercial license for CGAL does not cover this release of the 2D Triangulation package. You get this error, as you defined CGAL_LICENSE_ERROR."
 #      error "Your commercial license for CGAL does not cover this release \
        ^~~~~
In file included from /mnt/testsuite/include/CGAL/Triangulation_data_structure_3.h:28:0,
                 from /mnt/testsuite/include/CGAL/Triangulation_3.h:46,
                 from /mnt/testsuite/include/CGAL/Delaunay_triangulation_3.h:40,
                 from /home/cgal_tester/build/src/cmake/platforms/Fedora-rawhide-Release/test/Triangulation_3/test_delaunay_3.cpp:25:
/mnt/testsuite/include/CGAL/license/TDS_3.h:39:8: error: #error "Your commercial license for CGAL does not cover this release of the 3D Triangulation Data Structure package. You get this error, as you defined CGAL_LICENSE_ERROR."
 #      error "Your commercial license for CGAL does not cover this release \
        ^~~~~
```
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.10-Ic-168/Triangulation_3/TestReport_lrineau_Fedora-rawhide-Release.gz

**Comment by afabri** [comment]
I cannot reproduce the problem with VC++ on my machine.  @lrineau can you please have a look.

**Comment by lrineau** [comment]
`<CGAL/version_macros.h>` is not included, and `CGAL_FUTURE` is not expanded. Then in the preprocessor conditional, its value is  `0`.

**Comment by lrineau** [comment]
The Travis job has already detected an error, that is not a C++ error, but one reported by our automatic checking scripts:
```
./Scripts/developer_scripts/test_merge_of_branch HEAD
.. Testing permissions...
.. Testing licenses...
.. Testing CMake configuration...
OK
.. List of new files (please check that it is the expected list:
No new files.
.. Checking if all CMakeLists.txt are present...
.. Checking if all CMakeLists.txt project names are correct...
.. Checking include directives in header files...
The following files do not have an include directive of the package license header:
TDS_2/include/CGAL/internal/TDS_2/Edge_hash_function.h
TDS_2/include/CGAL/internal/TDS_2/edge_list.h
```
https://travis-ci.org/CGAL/cgal/jobs/219238437

The two files probably do not include the right license-checking header.

**Comment by lrineau** [comment]
And also the compilation error of `test/Mesh_2/test_lipschitz.cpp`:
https://travis-ci.org/CGAL/cgal/jobs/219238478#L4447

**Pull request merged**